src/jit/codegenxarch.cpp

```diff
@@ -1329,9 +1347,26 @@ void CodeGen::genCodeForDivMod(GenTreeOp* treeNode)
     }
     else
     {
-        // dividend must be in RAX
+#ifdef _TARGET_X86_
+        if (dividendIsLong)
+        {
+            assert(dividendLo != nullptr && dividendHi != nullptr);
+
+            // dividendLo must be in RAX; dividendHi must be in RDX
+            if (dividendLo->gtRegNum != REG_RAX)
+            {
+                inst_RV_RV(INS_mov, REG_RAX, dividendLo->gtRegNum, targetType);
+            }
+            if (dividendHi->gtRegNum != REG_RDX)
+            {
+                inst_RV_RV(INS_mov, REG_RDX, dividendLo->gtRegNum, targetType);
+            }
```

Comment:
Just below this we trash EDX before generating the div/idiv. How does that work?